packages/timeline-state-resolver/src/devices/telemetrics.ts

```diff
@@ -0,0 +1,168 @@
+	private errorMessage: string
+
+	private recentlySendCalls: Map<string, number> = new Map()
+	private retryConnectionTimer: Timer | undefined
+
+	constructor(deviceId: string, deviceOptions: DeviceOptionsTelemetrics, getCurrentTime: () => Promise<number>) {
+		super(deviceId, deviceOptions, getCurrentTime)
+
+		this.doOnTime = new DoOnTime(() => this.getCurrentTime())
+		this.handleDoOnTime(this.doOnTime, 'telemetrics')
+	}
+
+	get canConnect(): boolean {
+		return true
+	}
+
+	clearFuture(_clearAfterTime: number): void {
+		// No state to handle - we use a fire and forget approach
+	}
+
+	get connected(): boolean {
+		return this.statusCode === StatusCode.GOOD
+	}
+
+	get deviceName(): string {
+		return `${TELEMETRICS_NAME} ${this.deviceId}`
+	}
+
+	get deviceType() {
+		return DeviceType.TELEMETRICS
+	}
+
+	getStatus() {
+		const messages: string[] = []
+
+		switch (this.statusCode) {
+			case StatusCode.GOOD:
+				this.errorMessage = ''
+				messages.push('Connected')
+				break
+			case StatusCode.BAD:
+				messages.push('No connection')
+				break
+			case StatusCode.UNKNOWN:
+				this.errorMessage = ''
+				messages.push('Not initialized')
+				break
+		}
+
+		if (this.errorMessage) {
+			messages.push(this.errorMessage)
+		}
+
+		const deviceStatus: DeviceStatus = {
+			statusCode: this.statusCode,
+			messages,
+			active: this.isActive,
+		}
+		return deviceStatus
+	}
+
+	handleState(newState: TimelineState, mappings: Mappings): void {
+		super.onHandleState(newState, mappings)
+		_.each(newState.layers, (timelineObject, _layerName) => {
+			const telemetricsObject: TimelineObjTelemetrics = timelineObject as unknown as TimelineObjTelemetrics
+			telemetricsObject.content.presetShotIdentifiers.forEach((presetShotIdentifier: number) => {
+				const command = `P0C${presetShotIdentifier}\r`
+
+				if (this.hasCommandBeenSendWithinLastSecond(command, newState.time)) {
+					return
+				}
+
+				this.recentlySendCalls.set(command, newState.time)
+
+				this.doOnTime.queue(newState.time, undefined, () => {
+					this.socket.write(command)
+				})
+			})
+		})
+	}
+
+	private hasCommandBeenSendWithinLastSecond(command: string, time: number): boolean {
+		return this.recentlySendCalls.has(command) && this.recentlySendCalls.get(command)! >= time - 1000
+	}
+
+	async init(options: TelemetricsOptions): Promise<boolean> {
+		return Promise.resolve(this.setupSocket(options.host))
+	}
+
+	private async setupSocket(host: string): Promise<boolean> {
+		this.socket = new Socket()
+
+		this.socket.on('data', (data: Buffer) => {
+			this.emit('debug', `${this.deviceName} received data: ${data.toString()}`)
+		})
+
+		this.socket.on('error', (error: Error) => {
+			this.updateStatus(StatusCode.FATAL, error)
+		})
+
+		this.socket.on('close', (hadError: boolean) => {
+			this.doOnTime.dispose()
+			this.updateStatus(hadError ? StatusCode.FATAL : StatusCode.BAD)
+			this.reconnect(host)
+		})
+
+		this.socket.connect(SOCKET_PORT, host)
```

Comment:
I disagree. Since this file is called `telemetrics` and is a specific integration to Telemetrics and the Telemetrics device only support the use of port 5000, it adds an extra unnecessary layer of configuration, which is another potential error

Reply:
And there are plenty of scenarios this might be needed. Tunnels, proxies, e2e tests

Reply:
I can see the usecase if your network has configured it so that `someIpAddress:somePort` is routed to `telemetricsIp:telemetricsPort` and the telemetrics IP isn't exposed anywhere. 

I can be up for making the port an optional configuration that defaults to 5000 if no port is provided. I still think that, as of right now, this is extra configuration we don't need :)

Reply:
I have added an optional port to the TelemetricsOptions, so it's possible to connect to a different port. I would prefer if we for now don't expose a configuration for changing the port in Sofie